src/soilbiogeochem/SoilBiogeochemDecompCascadeBGCMod.F90

```diff
@@ -986,70 +974,43 @@ subroutine decomp_rate_constants_bgc(bounds, num_soilc, filter_soilc, &
          end do
       endif
 
-      if (use_vertsoilc) then
-         ! add a term to reduce decomposition rate at depth
-         ! for now used a fixed e-folding depth
-         do j = 1, nlevdecomp
-            do fc = 1,num_soilc
-               c = filter_soilc(fc)
-               depth_scalar(c,j) = exp(-zsoi(j)/decomp_depth_efolding)
-            end do
+      ! add a term to reduce decomposition rate at depth
+      ! for now used a fixed e-folding depth
+      do j = 1, nlevdecomp
+         do fc = 1, num_soilc
+            c = filter_soilc(fc)
+            if (use_vertsoilc) then
+               depth_scalar(c,j) = exp(-zsoi(j) / decomp_depth_efolding)
+            else
+               depth_scalar(c,j) = 1.0_r8
+            end if
```

Comment:
The above modification allowed me to eliminate a whole bunch of repetition in the loops below. (Similarly in the CN copy of this.)

Reply:
Nice.